app/code/Magento/InventorySalesAlert/Model/ProductIsSalable.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySalesAlert\Model;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\InventorySalesAlert\Api\ProductIsSalableInterface;
+use Magento\InventorySales\Model\StockResolver;
+use Magento\InventoryApi\Api\IsProductInStockInterface;
+use Magento\Framework\App\ObjectManager;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\Inventory\Model\IsProductInStock;
+
+class ProductIsSalable implements ProductIsSalableInterface
+{
+
+    /**
+     * @var \Magento\InventoryApi\Api\IsProductInStockInterface
+     */
+    private $stockItem;
+
+    /**
+     * @var \Magento\InventorySales\Model\StockResolver
+     */
+    private $stockResolver;
+
+    /**
+     * ProductIsSalable constructor.
+     *
+     * @param \Magento\InventorySales\Model\StockResolver         $stockResolver
+     * @param \Magento\InventoryApi\Api\IsProductInStockInterface $stockItem
+     */
+    public function __construct(
+        StockResolver $stockResolver = null,
+        IsProductInStockInterface $stockItem = null
+    ) {
+        $this->stockItem     = $stockItem ?: ObjectManager::getInstance()->get(IsProductInStock::class);
```

Comment:
This trick is needed only for backward compatibility, but i this case it is new code